core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala

```diff
@@ -414,6 +414,20 @@ trait DeltaSQLConfBase {
              |""".stripMargin)
       .booleanConf
       .createWithDefault(false)
+
+  val DELTA_LEGACY_ALLOW_AMBIGUOUS_PATHS =
+    buildConf("legacy.allowAmbiguousPaths")
+      .internal()
+      .doc("""
+             |Delta was unintentionally allowing CREATE TABLE queries with both 'delta.`path`'
```

Comment:
Delta was unintentionally allowing CREATE TABLE queries with both 'delta.`path`' and `LOCATION path` clauses. In the new version, we will raise an error for this case. This flag is added to allow users to skip the check. when it's set to true and there are two paths in CREATE TABLE, the `LOCATION path` clause will be ignored like what the old version does.